content/doc/book/system-administration/reverse-proxy-configuration-with-jenkins.adoc

```diff
@@ -0,0 +1,1411 @@
+Set the context path  when using the Ubuntu package by adding
+`+--prefix=/jenkins+` to `+JENKINS_ARGS+` in `+/etc/default/jenkins+` (
+or in `+/etc/sysconfig/jenkins+` for RHEL/CentOS package)
+
+When running on a dedicated server and you are using / as context, make
+sure you add a slash at the end of all URLs in proxy params in apache.
+Otherwise you might run into proxy errors. So
+
+[source]
+----
+ProxyPass / http://localhost:8080/ nocanon
+----
+
+instead of
+
+[source]
+----
+ProxyPass / http://localhost:8080 nocanon     # wont work
+----
+
+Note that this does *not* apply to the `+ProxyPassMatch+` directive,
+which behaves differently than `+ProxyPass+`.
+Below is an example of `+ProxyPassMatch+` to proxy all URLs other than
+`+/.well-known+` (a URL required by letsencrypt):
+
+[source]
+----
+ProxyPassMatch  ^/(?\!.well-known)  http://localhost:8080 nocanon
+----
+
+The _ProxyRequests Off_ prevents Apache from functioning as a forward
+proxy server (except for _ProxyPass_), it is advised to include it
+unless the server should function as a proxy.
+
+Both the `+nocanon+` option to `+ProxyPass+`, _and_
+`+AllowEncodedSlashes NoDecode+`, are required for certain Jenkins
+features to work.
+
+If you are running Apache on a Security-Enhanced Linux (SE-Linux)
+machine it is essential to make SE-Linux do the right thing by issuing
+as root
+
+[source]
+----
+setsebool -P httpd_can_network_connect true
+----
+
+If this is not issued Apache will not be allowed to forward proxy
+requests to Jenkins and only an error message will be displayed.
+
+Because Jenkins already compress its output, you can not use the normal
+proxy-html filter to modify urls:
+
+[source]
+----
+SetOutputFilter proxy-html
+----
+
+Instead you can use the following:
+
+[source]
+----
+SetOutputFilter INFLATE;proxy-html;DEFLATE
+ProxyHTMLURLMap http://your_server:8080/jenkins /jenkins
+----
+
+http://wiki.uniformserver.com/index.php/Reverse_Proxy_Server_2:_mod_proxy_html_2 +
+But since Jenkins seems to be well behaved it's even better to just not
+use SetOutputFilter and ProxyHTMLURLMap.
+
+If there are problems with Jenkins sometimes servicing random garbage
+pages, then the following may help:
+
+[source]
+----
+SetEnv proxy-nokeepalive 1
+----
+
+Some plug-ins determine URLs from client requests from Host header, so
+if you experience some problems with wrong URLs, you can try to switch
+on `+ProxyPreserveHost+` directive, which is switched off by default:
+
+[source]
+----
+ProxyPreserveHost On
+----
+
+===== mod_proxy with HTTPS
+
+If you'd like to run Jenkins with reverse proxy in HTTPS, one user
+reported that HTTPS needs to be terminated at Jenkins, not at the
+front-end Apache.
+See http://hudson.361315.n4.nabble.com/Hudson-behind-an-Apache-Reverse-Proxy-w-SSL-td370997.html[this
+e-mail thread] for more discussion.
+
+Note that you also may need to have access to your host via regular
+http, else the admin interface test will report a broken proxy setup.
```

Comment:
This is referencing an email conversation from 12 years ago.  It should be removed from the documentation.  There are several blog posts that give good hints that Jenkins and Apache with HTTPS is possible and relatively straighforward.  See:

* https://serverfault.com/questions/653408/correct-apache-reverse-proxy-config-with-ssl-for-jenkins-and-sonar
* https://www.agileana.com/blog/serve-jenkins-over-https-with-apache-as-proxy-and-certbot-letsencrypt-ssl/
* https://www.cloudbooklet.com/how-to-setup-jenkins-with-ssl-with-apache-reverse-proxy-on-ubuntu/
* https://mensfeld.pl/2013/06/jenkins-behind-apache-with-https-proxy-pass-with-ssl/

The oldest of those references is from 7 years ago.  The newest is from only 2 years ago.
```suggestion
```